modules/apim-adapter/src/main/java/com/axway/apim/api/specification/Swagger2xSpecification.java

```diff
@@ -98,13 +98,28 @@ public void configureBasePath(String backendBasePath, API api) throws AppExcepti
                     LOG.debug("Adding new host {}{} to Swagger-File based on backendBasePath: {}", url.getHost(), port, backendBasePath);
                     ((ObjectNode) swagger).put("host", url.getHost() + port);
                     LOG.info("Used the backendBasePath: {} to adjust host the API-Specification.", backendBasePath);
+                }else {
+                    if(swagger.get("host").asText().equals(url.getHost()+port)) {
+                        LOG.debug("Swagger Host: '"+swagger.get("host").asText()+"' already matches backendBasePath: '"+backendBasePath+"'. Nothing to do.");
+                    } else if (CoreParameters.getInstance().isOverrideSpecBasePath()){
```

Comment:
Don't change the spec unless overridebaseapth variable set as we can override the hostname via frontend APIs backend url.

Reply:
can you explain better?